packages/api/src/derives/fees/estimateFee.ts

```diff
@@ -0,0 +1,66 @@
+import {ApiInterfaceRx} from '@cennznet/api/types';
+import Extrinsic from '@cennznet/types/extrinsic/Extrinsic';
+import {ChargeTransactionPayment} from '@cennznet/types/runtime/transaction-payment';
+import {AnyAssetId, IExtrinsic} from '@cennznet/types/types';
+import {drr} from '@polkadot/rpc-core/rxjs';
+import {TypeRegistry} from '@polkadot/types';
+import {Address} from '@polkadot/types/interfaces';
+import {RuntimeDispatchInfo} from '@polkadot/types/interfaces/rpc';
+import ExtrinsicEra from '@polkadot/types/primitive/Extrinsic/ExtrinsicEra';
+import BN from 'bn.js';
+import {combineLatest, Observable, of} from 'rxjs';
+import {catchError, first, map, switchMap} from 'rxjs/operators';
+
+export function estimateFee(api: ApiInterfaceRx) {
+  /* To estimate fee correctly using extrinsic's fake signature */
+  return (extrinsic: IExtrinsic, userFeeAssetId: AnyAssetId, maxPayment?: string): Observable<any> => {
```

Comment:
have taken maxPayment as optional parameter, cause if the maxPayment value is different while fake signing and actually signing, then the fee estimated is not the same as deducted.

Reply:
Not a big deal and no need to change here, but I am thinking what if we need to add more options as arguments passed into the function?

Would be in favor of scoping args in a key-valued object whenever we need to add more than 2 args in a function, otherwise its quite hard for dev to know what is the meaning of each argument, for this one, its like:

```js
({
  extrinsic, 
  userFeeAssetId, 
  maxPayment
}) => {...}
```
```js
// Or
(extrinsic, userFeeAssetId, {
  maxPayment?: string,
  // more options comes here
}) => {... }
```
So that when this function is called, we as developer can easily identify what is what. Thats just my own opinion, there may be some other cases I did not get, welcome a discuss~